lib/matplotlib/backend_bases.py

```diff
@@ -2298,32 +2297,32 @@ def get_window_title(self):
             return self.manager.get_window_title()
 
     def set_window_title(self, title):
-        """
-        Set the title text of the window containing the figure.  Note that
-        this has no effect if there is no window (e.g., a PS backend).
+        """Set the title text of the window containing the figure.
+
+        This has no effect if there is no window (e.g., a PS backend).
         """
         if hasattr(self, "manager"):
             self.manager.set_window_title(title)
 
     def get_default_filename(self):
-        """
-        Return a string, which includes extension, suitable for use as
-        a default filename.
-        """
-        default_basename = self.get_window_title() or 'image'
-        default_basename = default_basename.replace(' ', '_')
+        """Return a suitable default filename, including the extension.
+        """
+        default_basename = self.get_window_title() or "image"
+        # Characters to be avoided in a NT path:
+        # https://msdn.microsoft.com/en-us/library/windows/desktop/aa365247(v=vs.85).aspx#naming_conventions
+        removed_chars = \
+            {"posix": " /\0", "nt": r'<>:"/\|?*\0'}.get(os.name, "_")
```

Comment:
Is the 'nt' case missing `' '`?

Reply:
@tacaswell if you have a look at #9056 you see that the user does not always want to shoot their own foot. Sometimes it is the interaction of several features, like in that case setting a title will affect the chosen name when you click the save button. And when matplotlib proposes a name which does not follow the naming conventions of an operating system, that is bad in my opinion.